elements/linuxmodule/fromdevice.cc

```diff
@@ -472,14 +471,19 @@ FromDevice::run_task(Task *)
     int npq = 0;
     Storage::index_type h;
     while (npq < _burst && (h = head()) != tail()) {
-	Packet *p = _queue[h];
+        Packet *p = _queue[h];
+        if (p == NULL) {
+            _task.fast_reschedule();
+            return npq > 0;
+	}
+        _queue[h] = 0;
 #if CLICK_DEBUG_SCHEDULING
-	emission_report(h);
+        emission_report(h);
 #endif
-	set_head(next_i(h));
-	output(0).push(p);
-	npq++;
-	_count++;
+        set_head(next_i(h));
+        output(0).push(p);
+        npq++;
+        _count++;
```

Comment:
It isn't super great when a commit includes both whitespace changes and technical changes.